streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -68,3 +83,20 @@ case class BatchInfo(
    */
   def numRecords: Long = streamIdToInputInfo.values.map(_.numRecords).sum
 }
+
+@DeveloperApi
+object BatchInfo {
+
+  /**
+   * Create `BatchInfo`. This is for binary compatibility.
+   */
+  def apply(
```

Comment:
Actually, this is a old public method that the compiler will create for case class.